src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -5,68 +5,106 @@
 
-            if (packages == null)
+            if (bondManifestDirectory == null)
             {
                 return Array.Empty<IOptionalSdkTemplatePackageInfo>();
             }
-            else
+
+            var manifestContent =
+                WorkloadManifest.LoadFromFolder(bondManifestDirectory.GetDirectories().Single().FullName);
+            var dotnetSdkTemplatePackages = new List<IOptionalSdkTemplatePackageInfo>();
+            foreach (var pack in manifestContent.SdkPackDetail)
             {
-                return packages
-                    .Select(l => new OptionalSdkTemplatePackageInfo(l)).ToArray();
+                if (pack.Value.kind.Equals("Template", StringComparison.OrdinalIgnoreCase))
+                {
+                    var optionalSdkTemplatePackageInfo = new OptionalSdkTemplatePackageInfo(pack.Key,
+                        pack.Value.version,
+                        Path.Combine(dotnetRootPath, "template-packs",
+                            pack.Key.ToLower() + "." + pack.Value.version.ToLower() + ".nupkg"));
+                    dotnetSdkTemplatePackages.Add(optionalSdkTemplatePackageInfo);
+                }
             }
-        }
 
-        public string DotnetSdkVersionUsedInVs()
-        {
-            return "5.1.100";
+            return dotnetSdkTemplatePackages;
         }
 
-        internal void SetDotnetSdkTemplatesLocation(DirectoryInfo directoryInfo)
+        public bool TryGetDotnetSdkVersionUsedInVs(string vsVersion, out string sdkVersion)
         {
-            _dotnetSdkTemplatesLocation = directoryInfo;
+            string dotnetExeDir = _netCoreSdkResolver.Value.GetDotnetExeDirectory();
+
+            if (!Version.TryParse(vsVersion, out var parsedVsVersion))
+            {
+                throw new ArgumentException(vsVersion + " is not a valid version");
+            }
+
+            // VS major minor version will match msbuild major minor
+            // and for resolve SDK, major minor version is enough
+            var msbuildMajorMinorVersion = new Version(parsedVsVersion.Major, parsedVsVersion.Minor, 0);
+
+            var resolverResult =
+                _netCoreSdkResolver.Value.ResolveNETCoreSdkDirectory(null, msbuildMajorMinorVersion, true, dotnetExeDir);
+
+            if (resolverResult.ResolvedSdkDirectory == null)
+            {
+                sdkVersion = null;
+                return false;
+            }
+            else
+            {
+                sdkVersion = new DirectoryInfo(resolverResult.ResolvedSdkDirectory).Name;
+                return true;
+            }
         }
 
         private class OptionalSdkTemplatePackageInfo : IOptionalSdkTemplatePackageInfo
         {
-            public OptionalSdkTemplatePackageInfo(LocalPackageInfo localPackageInfo)
+            public OptionalSdkTemplatePackageInfo(string templatePackageId, string templateVersion, string path)
```

Comment:
minor: consider overriding ```ToString``` for POCOs

Reply:
no serialization code for type that does not need serialization